ddtrace/debugging/_probe/model.py

```diff
@@ -83,6 +84,22 @@ def _(self):
             raise_on_exceed=False,
         )
 
+    def update(self, other):
+        # type: (Probe) -> None
+        """Update the mutable fields from another probe."""
+        if self.probe_id != other.probe_id:
+            log.error("Probe ID mismatch when updating mutable fields")
+            return
+
+        if self.version == other.version:
+            return
+
+        for attrib in self.__attrs_attrs__:
+            setattr(self, attrib.name, getattr(other, attrib.name))  # type: ignore[attr-defined]
```

Comment:
There are fields that are not mutable, this looks like it's overwriting everything. Things like `probe_type` and `language` should not be overwritten.